.github/workflows/publish-pypi.yml

```diff
@@ -70,3 +70,38 @@ jobs:
         # Only publish to real PyPI on release
         if: github.event_name == 'release'
         uses: pypa/gh-action-pypi-publish@release/v1
+  sign-files:
+    name: >-
+      Sign the Python 🐍 distribution 📦 with Sigstore
+      and upload them to GitHub Release
+    if: github.repository_owner == 'pyopensci'
+    needs:
+    - publish
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write  # this permission is mandatory for modifying GitHub Releases
+      id-token: write  # this permission is mandatory for sigstore
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Sign the dists with Sigstore
+      uses: sigstore/gh-action-sigstore-python@v2.1.1
+      with:
+        inputs: >-
+          ./dist/*.tar.gz
+          ./dist/*.whl
+    - name: Upload artifact signatures to GitHub Release
+      # Only upload on release
+      if: github.event_name == 'release'
```

Comment:
It probably makes sense to put this into the job conditional. Otherwise, this may end up signing undesired files, which may be used in supply chain attacks.

Reply:
Done